lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -166,19 +166,19 @@ def verify(payment, options = {})
 
       def application_fee_from_response(response)
         return unless response.success?
-
-        application_fees = response.params["data"].select { |fee| fee["object"] == "application_fee" }
-        application_fees.first["id"] unless application_fees.empty?
+        response.params["application_fee"] unless response.params["application_fee"].empty?
       end
 
       def refund_application_fee(money, identification, options = {})
         return Response.new(false, "Application fee id could not be found") unless identification
 
         post = {}
         add_amount(post, money, options)
-        options.merge!(:key => @fee_refund_api_key)
+        options.merge!(:key => @fee_refund_api_key) if @fee_refund_api_key
+        options.delete(:stripe_account) if options[:stripe_account]
 
-        commit(:post, "application_fees/#{CGI.escape(identification)}/refund", post, options)
+        refund_fee = commit(:post, "application_fees/#{CGI.escape(identification)}/refunds", post, options)
+        refund_fee.success? ? refund_fee : Response.new(false, "Application fee could not be refunded: #{refund_fee.message}")
```

Comment:
There may be a better way to hand this and 468. Their current state is based on the preexisting line 173 above.

Reply:
Agreed. There is a quite a bit going on in just a few lines of code. At a minimum, while I try to load up in my head what's going on, `refund_fee.success? ? refund_fee : Response.new(false, "Application fee could not be refunded: #{refund_fee.message}")` could be moved to a private method as the same pattern is used on line 468.

Reply:
Yeah, I'm with @dtykocki. Maybe something like

```ruby
def require_success!(obj, message)
  obj.success? ? obj : Response.new(false, message)
end
```

which'd make it clearer what's going on here.